tests/Builder/Stage/OutStageTest.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Tests\Builder\Stage;
+
+use MongoDB\Builder\Accumulator;
+use MongoDB\Builder\Expression;
+use MongoDB\Builder\Pipeline;
+use MongoDB\Builder\Stage;
+use MongoDB\Tests\Builder\PipelineTestCase;
+
+use function MongoDB\object;
+
+/**
+ * Test $out stage
+ */
+class OutStageTest extends PipelineTestCase
+{
+    public function testOutputToADifferentDatabase(): void
+    {
+        $pipeline = new Pipeline(
+            Stage::group(
+                _id: Expression::stringFieldPath('author'),
+                books: Accumulator::push(
+                    Expression::stringFieldPath('title'),
+                ),
+            ),
+            Stage::out(
+                object(
+                    db: 'reporting',
+                    coll: 'authors',
+                ),
+            ),
```

Comment:
I dislike this signature, as passing an object seems overly complex. This is one of those cases where a separate (not auto-generated) encoder would work better, so we can essentially have the following signature:
```php
function out(string $coll, ?string $db = null)
```

Reply:
Related to https://github.com/mongodb/mongo-php-builder/pull/56#discussion_r1466456462